src/module/item/effect/document.ts

```diff
@@ -74,8 +75,10 @@ class EffectPF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extends Ab
     /** Increases if this is a counter effect, otherwise ignored outright */
     async increase(): Promise<void> {
         const badge = this.system.badge;
+
         if (badge?.type === "counter" && !this.isExpired) {
-            const value = badge.value + 1;
+            const shouldLoop = badge.loop && badge.labels && badge.value >= badge.max;
```

Comment:
You caught me before I could edit it, but also remove the && badge.labels. badge.loop should be false if there are no labels anyways.

Reply:
Did quick test and yeah, that appears to be the case. Commit fixed.